src/psyclone/domain/lfric/lfric_constants.py

```diff
@@ -374,6 +373,44 @@ def __init__(self):
             # Logging module (used for runtime checks)
             "logging": {"module": "log_mod"}}
 
+    @staticmethod
+    def specific_function_space(name):
+        '''
+        Maps from a valid kernel metadata function-space name to one
+        that exists within the LFRic infrastructure. This is necessary
+        because meta-data can contain 'wildcard' names such as 'any_w2' but,
+        when generating code, we need the name of a specific function space
+        that is recognised by the LFRic infrastructure.
+
+        :param str name: the name of the function space in metadata.
+
+        :returns: the name of a specific function space.
+        :rtype: str
+
+        :raises ValueError: if the supplied name is not a valid LFRic \
+                            function-space name.
+        :raises InternalError: if an unrecognised wildcard function-space \
+                               name is supplied.
+        '''
+        space = name.lower()
+        if space not in LFRicConstants.VALID_FUNCTION_SPACE_NAMES:
+            raise ValueError(
+                f"'{space}' is not a recognised LFRic function space (one of "
+                f"{LFRicConstants.VALID_FUNCTION_SPACE_NAMES}).")
+
+        if not space.startswith("any_"):
+            return space
+        elif space == "any_w2":
+            return "w2"
```

Comment:
These should be configurable at some point. Reference the config issue I suggested earlier?

Reply:
While it's true, they could be configurable, there's no pressing need to do so so I don't see the value of an issue?

Reply:
I've changed my mind about this and created #1709. I'll update the code with a TODO.